metricbeat/module/vsphere/cluster/cluster.go

```diff
@@ -0,0 +1,174 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package cluster
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/elastic/beats/v7/metricbeat/mb"
+	"github.com/elastic/beats/v7/metricbeat/module/vsphere"
+
+	"github.com/vmware/govmomi"
+	"github.com/vmware/govmomi/property"
+	"github.com/vmware/govmomi/view"
+	"github.com/vmware/govmomi/vim25/mo"
+)
+
+// init registers the MetricSet with the central registry as soon as the program
+// starts. The New function will be called later to instantiate an instance of
+// the MetricSet for each network is defined in the module's configuration. After the
+// MetricSet has been created then Fetch will begin to be called periodically.
+func init() {
+	mb.Registry.MustAddMetricSet("vsphere", "cluster", New,
+		mb.WithHostParser(vsphere.HostParser),
+		mb.DefaultMetricSet(),
+	)
+}
+
+// MetricSet type defines all fields of the MetricSet.
+type ClusterMetricSet struct {
+	*vsphere.MetricSet
+}
+
+type assetNames struct {
+	outputNtNames []string
+	outputDsNames []string
+	outputHsNames []string
+}
+
+// New creates a new instance of the MetricSet.
+func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
+	ms, err := vsphere.NewMetricSet(base)
+	if err != nil {
+		return nil, err
+	}
+	return &ClusterMetricSet{ms}, nil
+}
+
+// Fetch methods implements the data gathering and data conversion to the right
+// format. It publishes the event which is then forwarded to the output. In case
+// of an error set the Error field of mb.Event or simply call report.Error().
+func (m *ClusterMetricSet) Fetch(ctx context.Context, reporter mb.ReporterV2) error {
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	client, err := govmomi.NewClient(ctx, m.HostURL, m.Insecure)
+	if err != nil {
+		return fmt.Errorf("error in NewClient: %w", err)
+	}
+	defer func() {
+		if err := client.Logout(ctx); err != nil {
+			m.Logger().Debug(fmt.Errorf("error trying to logout from vshphere: %w", err))
+		}
+	}()
+
+	c := client.Client
+
+	// Create a view of Cluster objects
+	mgr := view.NewManager(c)
+
+	v, err := mgr.CreateContainerView(ctx, c.ServiceContent.RootFolder, []string{"ClusterComputeResource"}, true)
+	if err != nil {
+		return fmt.Errorf("error in CreateContainerView: %w", err)
+	}
+
+	defer func() {
+		if err := v.Destroy(ctx); err != nil {
+			m.Logger().Errorf("error trying to destroy view from vshphere: %w", err)
+		}
+	}()
+
+	// Retrieve summary property for all Clusters
+	var clt []mo.ClusterComputeResource
+	err = v.Retrieve(ctx, []string{"ClusterComputeResource"}, []string{}, &clt)
```

Comment:
How about just collecting what we need here by providing list in `[]string{}`.
e.g.
```suggestion
	err = v.Retrieve(ctx, []string{"ClusterComputeResource"}, []string{"name", "configuration"}, &clt)
```